pkg/operator/controller/status/controller.go

```diff
@@ -685,3 +729,45 @@ func operatorStatusesEqual(a, b configv1.ClusterOperatorStatus) bool {
 
 	return true
 }
+
+// joinConditions merges two cluster operator conditions into one.
+// If both conditions have the same status:
+// - Reason becomes: "MultipleComponents[Not]" + condType.
+// - Messages are concatenated.
+// If the new condition has a higher-priority status, it overrides the current one.
+// Status priority (highest to lowest): True, False, Unknown, Empty.
+func joinConditions(currCond, newCond configv1.ClusterOperatorStatusCondition, condType string) configv1.ClusterOperatorStatusCondition {
+	switch newCond.Status {
+	case configv1.ConditionTrue:
+		if currCond.Status == configv1.ConditionTrue {
+			currCond.Reason = "MultipleComponents" + condType
+			currCond.Message += newCond.Message
+		} else {
+			// Degraded=True status overrides other statuses.
+			currCond.Status = newCond.Status
+			currCond.Reason = newCond.Reason
+			currCond.Message = newCond.Message
+		}
+	case configv1.ConditionFalse:
+		if currCond.Status == configv1.ConditionFalse {
+			currCond.Reason = "MultipleComponentsNot" + condType
+			currCond.Message += newCond.Message
+		} else if currCond.Status != configv1.ConditionTrue {
+			// Degraded=False overrides unknown and empty statuses.
```

Comment:
This seems wrong.  If I have one component with `Degraded=Unknown` and one with `Degraded=False`, then doesn't that mean that the combined status is unknown?  

For example, if my load balancer is ready but status of DNS is unknown, then I should say that the overall status is `Degraded=Unknown`, not `Degraded=False`.

Reply:
I think you are right, if any of sources reports `Unknown` and others don't report `Degraded=True` the whole status should become `Unknown`.

Reply:
Done.